### PR TITLE
[KOGITO-5394] Workitem error handling strategy example

### PR DESCRIPTION
**Pull request by dmarrazzo**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `main` branch!**

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

File: process-error-handling/src/main/java/org/acme/wih/CustomTaskWorkItemHandler.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            manager.completeWorkItem(workItem.getStringId(), results);
+        }
+
+        LOG.debug("end");
+    }
+
+    private void handleError(String strategy) {
+        throw new ProcessWorkItemHandlerException("error_handling",
+                ProcessWorkItemHandlerException.HandlingStrategy.valueOf(strategy),
+                new Throwable(strategy + " strategy test"));
```
Comment (fjtirado) [inline process-error-handling/src/main/java/org/acme/wih/CustomTaskWorkItemHandler.java]: Throwable is too heavy ;), lets use new IllegalStateException

File: process-error-handling/src/main/java/org/acme/wih/CustomTaskWorkItemHandler.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    @Override
+    public void abortWorkItem(KogitoWorkItem workItem, KogitoWorkItemManager manager) {
+        LOG.debug("nope");
```
Comment (fjtirado) [inline process-error-handling/src/main/java/org/acme/wih/CustomTaskWorkItemHandler.java]: Lets use something more descriptive than "nope", something like "ABORTED!!!!!"

**Review by fjtirado — APPROVED** [review]
Superb job Donato
Thanks a lot

File: process-error-handling/README.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+```json
+{
+  "id": "ae4a5fd4-3d49-4ba5-92dd-4b6222492fdd",
+  "message": "Hello john"
+}
+```
+
+### Probe the RETRY strategy
+
+**TIP:** In the following example, we are going to show the `curl` command line. Although, a more comfortable way is using the following [REST client](https://https://marketplace.visualstudio.com/items?itemName=humao.rest-client), you can find in the project the file with all the requests: `src/test/http/error-handling.http`.
```
Comment (ricardozanini) [inline process-error-handling/README.md]:
```suggestion
> **TIP:** In the following example, we are going to show the `curl` command line. Although, a more comfortable way is using the following [REST client](https://https://marketplace.visualstudio.com/items?itemName=humao.rest-client), you can find in the project the file with all the requests: `src/test/http/error-handling.http`.
```

File: process-error-handling/pom.xml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    </dependencies>
+  </dependencyManagement>
+  <dependencies>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-resteasy</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.kie.kogito</groupId>
+      <artifactId>kogito-quarkus</artifactId>
```
Comment (ricardozanini) [inline process-error-handling/pom.xml]:
```suggestion
      <artifactId>kogito-quarkus-processes</artifactId>
```

**Review by ricardozanini — APPROVED** [review]
Nice example! Many thanks!

@fjtirado, do you believe we can do a version of this example for SW? The retries/error handling implementation you added does support this use case, I believe.

**Comment by dmarrazzo** [comment]
Thanks @fjtirado @ricardozanini 
I squashed the changes in the new commit (TBH I'm not familiar with PR workflow, hope I've done right)

**Comment by ricardozanini** [comment]
Jenkins retest this

**Comment by ricardozanini** [comment]
@dmarrazzo as soon as the CI is green we can merge it. :)

**Pull request merged**